ArmoryQt.py

```diff
@@ -5212,13 +5220,42 @@ def unregisterProgressCallback(self, id):
 ############################################
 
 if 1:
-   #setup splash screen
+   # 1) Start the bridge BEFORE anything else (singleton) so dialogs can query it
+   #    Start with a no-op callback; we'll wire readiness to the dialog instance
+   #    below to avoid overwriting global callbacks.
+   TheBDM.startBridge(getBridgeArgList(), lambda *args, **kwargs: None)
+
+   #    Show Setup Manager (modal and focused)
+   #    Bridge is already running; dialog must not try to start it
+   from qtdialogs.DlgSetupManager import DlgSetupManager
+
+   dlg = DlgSetupManager(parent=None, main=None)
+   try:
+      from armoryengine.CppBridge import TheBridge
+      if TheBridge.bridgeSocket.bip15xConnection.ready():
+         QtCore.QMetaObject.invokeMethod(
+            dlg, 'onBridgeReady', QtCore.Qt.QueuedConnection)
+      else:
+         TheBridge.bridgeSocket.bip15xConnection.setNotifyReadyLbd(
+            lambda: QtCore.QMetaObject.invokeMethod(
+               dlg, 'onBridgeReady', QtCore.Qt.QueuedConnection))
+   except Exception:
+      pass
+
+   if dlg.exec_() != QtWidgets.QDialog.Accepted:
+      try:
+         from armoryengine.CppBridge import TheBridge
+         TheBridge.service.shutdown()
+      except Exception:
+         pass
```

Comment:
Dont handle exceptions silently. If you do not know what to do with the exception, they don't catch it at all